packages/docs/docs/tutorials/bots/hl7-into-fhir.md

```diff
@@ -0,0 +1,120 @@
+
+* Make sure you have an account on Medplum, if not, [register](https://app.medplum.com/register).
+* Create a [ClientApplication](https://app.medplum.com/admin/project) on Medplum called "ADT Bot Client Application".
+* (Optional) Create a very restrictive [AccessPolicy](https://app.medplum.com/AccessPolicy) called "ADT Bot Policy", make it so that the policy only allows readwrite on the Patient object.
+* (Optional) In the [ProjectAdmin dashboard](https://app.medplum.com/admin/project) apply the "ADT Bot Policy" policy to the `ClientApplication` by clicking `Access`.
+
+### Bot Setup
+
+* Linking the ADT feed to Medplum is done through [Medplum Bots](https://app.medplum.com/Bot).
+* At a high level, each Bot exposes an endpoint and HL7 messages are posted over HTTPS to that endpoint.
+* The HL7 message is parsed and then converted to a corresponding FHIR object.  
+* (Optional) If needed, you can link a [Subscription](https://app.medplum.com/Subscription) to the FHIR objects that the Bot creates to notify downstream applications that new data is available.
+
+* Make the Easypost subscription bot that will listen for webhooks
+  * First, [create a bot](https://app.medplum.com/admin/project) called ADT Handler Bot and save it
+  * Paste the code below into the Bot you created and save.
+
+  ```js
+  export async function handler(medplum, event) {
+    const input = event.input;
+    //Log Message Type
+    const messageType = input.get('MSH').get(8); 
+    console.log(messageType); 
+
+    //Get patient name
+    const givenName = input.get('EVN').get(5).get(1);
+    const familyName = input.get('EVN').get(5).get(2);
+
+    //Get patient ID
+    const mrnNumber = input.get('PID').get(3).get(4);
+
+    var patient = await medplum.searchOne('Patient?identifier=' + mrnNumber);
+
+    if (patient) {
+        console.log('Patient already in the system');
+    } else {
+    patient = await medplum.createResource({
+    resourceType: 'Patient',
+    name: [
+        {
+        given: [givenName],
+        family: familyName,
+        },
+    ],
+    identifier: [{ system: 'www.myhospitalsystem.org/IDs', value: mrnNumber }]
+    });
+    console.log('Created patient', patient.id);
+    }
+
+    //Based on the messageType, you may consider making additional FHIR objects here
+
+    //Return Ack 
+    return input.buildAck().toString();s
+    }
+  ```
+
+HL7 messages have many nuances that are specific to the organization that is publishing them, so it is highly recommended to get some documentation and sample messages to ensure that you are capturing all the data you want to.
+
+### Testing your Bot
+
+You'll need your bot id (see [bot List](https://app.medplum.com/Bot) and click) to execute the bot. Once you have found it, you can attempt to execute your Bot using an HTTP message by sending the following via curl.  Note the content type.
+
+```cURL
+curl --location --request POST 'https://api.medplum.com/fhir/R4/Bot/<bot-id>/$execute' \
```

Comment:
Remove `--location`

I think most people would be more familiar with `-x POST` rather than `--request POST`, but that's stylistic.